pgjdbc/src/main/java/org/postgresql/PGConnection.java

```diff
@@ -244,6 +244,19 @@ public interface PGConnection {
    */
   PGReplicationConnection getReplicationAPI();
 
+  /**
+   * Change a user's password to the specified new password.
+   * If the specific encryption type is not specified, this method defaults to querying the database server for the server's default password_encryption.
+   * This method does not send the new password in plain text to the server.
+   * Instead, it encrypts the password locally and sends the encoded hash so that the plain text password is never sent on the wire.
+   *
+   * @param user The username of the database user
+   * @param newPassword The new password for the database user
+   * @param encryptionType The type of password encryption to use or null if the database server default should be used.
+   * @throws SQLException If the password could not be altered
+   */
+  void alterUserPassword(String user, String newPassword, @Nullable String encryptionType) throws SQLException;
```

Comment:
Can the implementation move to `default`?
Have you considered using `char[]` instead of `String` for the password?

Reply:
default ?

Reply:
> Can the implementation move to default?

Pretty sure he means adding a default implementation to the interface.

Was going to do that but the `PGConnection` only has our own methods and does not inherit from `java.sql.Connection` so would have had to have a raw cast like:

```java
void alterUserPassword(String user, String newPassword, @Nullable String encryptionType) throws SQLException {
  Connection conn = (Connection) this;
  // ...  
}
```

That'd probably work fine but felt weird. On the plus side that does get rid of having to add it the dummy implementation to that one test implementation.

Thoughts?

> Have you considered using char[] instead of String for the password?

Yes I had brought that up in the original PR as well. I'll take a look at that again, I think I had ended up using `String` because of that was the signature of one of the nested calls but that might have been from the first version of this.

Reply:
> On the plus side that does get rid of having to add it the dummy implementation to that one test implementation

We target Java 8, so using default implementation in interface sounds a right thing to do.

> That'd probably work fine but felt weird

I think it should be good enough.